### ISSUE TITLE
对于处理过程数据shape的疑问

### ISSUE THREAD

**Issue by RKO-max** [gudaochangsheng/MaskUnet#7]
你好，感谢你开源这样一个出色的工作！然后我想问一下，在training-baed的方法中，微调的图片resolution好像是[N,3,512,512]是吧，是否需要保持和预训练unet的数据集一样的shape，以及经过你使用的VAE后，图片会变成怎么样的一个shape的latent，因为我想要用自己的数据集以及自己一个VAE来尝试你这个工作，所以想要确认一些细节。

**Comment by RKO-max** [comment]
我看到你里面的数据是：
sample = torch.randn(1, 4, 64, 64).to(accelerator.device)
timestep = torch.tensor([50]).to(accelerator.device)
encoder_hidden_states = torch.randn(1, 77, 768).to(accelerator.device)
那如果我的数据是sample[1,32,16,16]，encoder_hidden_states是[1,256,64]，那在train_hyperunet中前面具体有哪些参数需要修改呢，希望作者可以帮忙指明！

**Comment by gudaochangsheng** [comment]
> 你好，感谢你开源这样一个出色的工作！然后我想问一下，在training-baed的方法中，微调的图片resolution好像是[N,3,512,512]是吧，是否需要保持和预训练unet的数据集一样的shape，以及经过你使用的VAE后，图片会变成怎么样的一个shape的latent，因为我想要用自己的数据集以及自己一个VAE来尝试你这个工作，所以想要确认一些细节。

There is no need to normalize the shape, this is the normalized code at https://github.com/gudaochangsheng/MaskUnet/blob/d342b13fcdd063f3d633d37b3016aabddbe32353/training/train_hyperunet.py#L850. 

It will change like this [N,3,512,512]-->[N,4,64,64].

**Comment by gudaochangsheng** [comment]
> 我看到你里面的数据是： sample = torch.randn(1, 4, 64, 64).to(accelerator.device) timestep = torch.tensor([50]).to(accelerator.device) encoder_hidden_states = torch.randn(1, 77, 768).to(accelerator.device) 那如果我的数据是sample[1,32,16,16]，encoder_hidden_states是[1,256,64]，那在train_hyperunet中前面具体有哪些参数需要修改呢，希望作者可以帮忙指明！

This input is noise, not the dataset input.

**Comment by RKO-max** [comment]
> > > > 我看到你里面的数据是： sample = torch.randn(1, 4, 64, 64).to(accelerator.device) timestep = torch.tensor([50]).to(accelerator.device) encoder_hidden_states = torch.randn(1, 77, 768).to(accelerator.device) 那如果我的数据是sample[1,32,16,16]，encoder_hidden_states是[1,256,64]，那在train_hyperunet中前面具体有哪些参数需要修改呢，希望作者可以帮忙指明！
> > > 
> > > 
> > > This input is noise, not the dataset input.
> > 
> > 
> > Here, the input noise is just a simulation？because I saw in the train_hyperunet.py file in the project that there is <img alt="Image" width="982" height="546" src="https://private-user-images.githubusercontent.com/200892160/466290007-8b80e385-d618-483e-930b-4f41ba07444e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI1NzI0NTEsIm5iZiI6MTc1MjU3MjE1MSwicGF0aCI6Ii8yMDA4OTIxNjAvNDY2MjkwMDA3LThiODBlMzg1LWQ2MTgtNDgzZS05MzBiLTRmNDFiYTA3NDQ0ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcxNVQwOTM1NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kY2UyOTgwNTllOTIxNDRmNGI3NzY4ZWI0ZWQ0OGE5YzZiNzNlNWM4ZTYxYWY5YWU1NTk2ZDQzNTRlZGZjNTAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3uEgsXOE5db-NKn4ipw79OBZ3AxwRK-UHtjVzizZn-8"> The latent data inside seems to be the dataset input. During subsequent training, this latent data is added with noise and then used with UNET to obtain model_pred. Therefore, I think we should still use the dataset input during formal training.
> 
> I understand, but after VAE, the number of channels becomes 4. I don't understand how your data dimension comes from. According to your logic, did you train VAE yourself?

Yes，I have a pre-trained VAE, it can encode the data([N,3,256,256]) into [N,32,16,16]，so my data actually is the data encoded by the VAE

**Comment by gudaochangsheng** [comment]
> > > > > 我看到你里面的数据是： sample = torch.randn(1, 4, 64, 64).to(accelerator.device) timestep = torch.tensor([50]).to(accelerator.device) encoder_hidden_states = torch.randn(1, 77, 768).to(accelerator.device) 那如果我的数据是sample[1,32,16,16]，encoder_hidden_states是[1,256,64]，那在train_hyperunet中前面具体有哪些参数需要修改呢，希望作者可以帮忙指明！
> > > > 
> > > > 
> > > > This input is noise, not the dataset input.
> > > 
> > > 
> > > Here, the input noise is just a simulation？because I saw in the train_hyperunet.py file in the project that there is <img alt="Image" width="982" height="546" src="https://private-user-images.githubusercontent.com/200892160/466290007-8b80e385-d618-483e-930b-4f41ba07444e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI1NzI0NTEsIm5iZiI6MTc1MjU3MjE1MSwicGF0aCI6Ii8yMDA4OTIxNjAvNDY2MjkwMDA3LThiODBlMzg1LWQ2MTgtNDgzZS05MzBiLTRmNDFiYTA3NDQ0ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcxNVQwOTM1NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kY2UyOTgwNTllOTIxNDRmNGI3NzY4ZWI0ZWQ0OGE5YzZiNzNlNWM4ZTYxYWY5YWU1NTk2ZDQzNTRlZGZjNTAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3uEgsXOE5db-NKn4ipw79OBZ3AxwRK-UHtjVzizZn-8"> The latent data inside seems to be the dataset input. During subsequent training, this latent data is added with noise and then used with UNET to obtain model_pred. Therefore, I think we should still use the dataset input during formal training.
> > 
> > 
> > I understand, but after VAE, the number of channels becomes 4. I don't understand how your data dimension comes from. According to your logic, did you train VAE yourself?
> 
> Yes，I have a pre-trained VAE, it can encode the data([N,3,256,256]) into [N,32,16,16]，so my data actually is the data encoded by the VAE

Maybe you can change this https://github.com/gudaochangsheng/MaskUnet/blob/d342b13fcdd063f3d633d37b3016aabddbe32353/training/train_hyperunet.sh#L10 -->256
